File: app/src/main/java/com/stedi/randomimagegenerator/app/other/logger/SoutLogger.java
package com.stedi.randomimagegenerator.app.other.logger;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public class SoutLogger implements Logger {
    private final String tag;

    public SoutLogger(@NonNull String tag) {
        this.tag = tag;
    }

    @Override
    public void log(@NonNull Object from, @Nullable String message) {
        System.out.println(tag + ": " + from.getClass().getSimpleName() + ": " + message);
    }

    @Override
    public void log(@NonNull Object from, @Nullable Throwable t) {
        System.out.println(tag + ": " + from.getClass().getSimpleName() + ": Throwable: ");
        if (t != null)
            t.printStackTrace();
    }

    @Override
    public void log(@NonNull Object from, @Nullable String message, @Nullable Throwable t) {
        System.out.println(tag + ": " + from.getClass().getSimpleName() + ": " + message + "; Throwable: ");
        if (t != null)
            t.printStackTrace();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/other/logger/SoutLogger.kt
package com.stedi.randomimagegenerator.app.other.logger

class SoutLogger(private val tag: String) : Logger {

    override fun log(from: Any, message: String?) {
        println(tag + ": " + from.javaClass.simpleName + ": " + message)
    }

    override fun log(from: Any, t: Throwable?) {
        println(tag + ": " + from.javaClass.simpleName + ": Throwable: ")
        t?.printStackTrace()
    }

    override fun log(from: Any, message: String?, t: Throwable?) {
        println(tag + ": " + from.javaClass.simpleName + ": " + message + "; Throwable: ")
        t?.printStackTrace()
    }
}